modules/launcher/src/main/java/org/mule/module/launcher/coreextension/DefaultMuleCoreExtensionManager.java

```diff
@@ -154,4 +165,16 @@ public void setPluginClassLoaderManager(PluginClassLoaderManager pluginClassLoad
     {
         this.pluginClassLoaderManager = pluginClassLoaderManager;
     }
+
+    /**
+     * Creates a {@link AdaptedDeploymentListener}.
+     *
+     * @param artifactDeploymentListener the artifactDeploymentListener to be adapted.
+     * @param type: the artifact type.
+     * @return an AdaptedDeploymentListener.
+     */
+    AdaptedDeploymentListener getAdaptedArtifactDeploymentListener (ArtifactDeploymentListener artifactDeploymentListener, ArtifactType type)
```

Comment:
Always use interfaces if possible. In this case, and in the test case, use DeploymentListener